core/src/main/java/io/questdb/cairo/wal/WalApplySqlExecutionContext.java

```diff
@@ -36,13 +36,18 @@
 class WalApplySqlExecutionContext extends SqlExecutionContextImpl {
     private TableToken tableToken;
 
-    WalApplySqlExecutionContext(CairoEngine cairoEngine, int workerCount, int sharedWorkerCount) {
-        super(cairoEngine, workerCount, sharedWorkerCount);
+    WalApplySqlExecutionContext(CairoEngine engine, int workerCount, int sharedWorkerCount) {
+        super(engine, workerCount, sharedWorkerCount);
+        if (!engine.getConfiguration().isMatViewParallelSqlEnabled()) {
```

Comment:
WAL apply does not run selects, this change looks confusing.

Reply:
But it runs SQL, e.g. UPDATEs or INSERTs AS SELECTs.

Reply:
It does not run insert as as selects, they are processed as inserts from WAL apply point of view. Add a comment please why it can run multithreaded SQL

Reply:
Added a comment in 69147e7bac1f0f53bf9e996362fc9438956a786f